examples/oauth.rs

```diff
@@ -187,35 +186,20 @@ async fn login_authorized(
     let cookie = format!("{}={}; SameSite=Lax; Path=/", COOKIE_NAME, cookie);
 
     // Set cookie
-    let r = http::Response::builder()
-        .header("Location", "/")
-        .header(SET_COOKIE, cookie)
-        .status(302);
+    let mut headers = HeaderMap::new();
+    headers.insert(SET_COOKIE, cookie.parse().unwrap());
 
-    r.body(Body::empty()).unwrap()
-}
-
-// Utility to save some lines of code
-struct Redirect(String);
-impl IntoResponse for Redirect {
-    type Body = Body;
-    type BodyError = hyper::Error;
-
-    fn into_response(self) -> http::Response<Body> {
-        let builder = http::Response::builder()
-            .header("Location", self.0)
-            .status(StatusCode::FOUND);
-        builder.body(Body::empty()).unwrap()
-    }
+    (headers, Redirect::permanent("/".parse().unwrap()))
 }
 
 struct AuthRedirect;
+
 impl IntoResponse for AuthRedirect {
-    type Body = Body;
-    type BodyError = hyper::Error;
+    type Body = Empty<Bytes>;
+    type BodyError = <Self::Body as axum::body::HttpBody>::Error;
 
-    fn into_response(self) -> http::Response<Body> {
-        Redirect("/auth/discord".to_string()).into_response()
+    fn into_response(self) -> http::Response<Self::Body> {
+        Redirect::permanent("/auth/discord".parse().unwrap()).into_response()
```

Comment:
I'm not sure if ´308´ is the best status code for a redirect like this. To my knowledge that would tell search engines that http:://example.com/auth/discord no longer exists and it will only index the page it points to. Maybe that only matters in a frontend-setting though now that I think about it.

Reply:
The more important thing is browsers may cache this redirect. I don't think I've ever seen permanent redirects in auth code.

Reply:
they can happen but should not happen, however, the original tower version had a version where you can provide the status code - based on what you feel is best as long as this is still an option the base return is not as important. 

In most systems, I have used and built we mapped 308 -> 301 as they are essentially the same and most don't check for 308.  

side note: 301 you can change the method in the redirect -> 308 methods is persisted so in discord example 308 is 100% correct as it persists the method.

Reply:
I think a 302 Found is fine to use aswell

Reply:
I've changed it to 302 Found. Thats also what the example used originally actually.